contracts/Sales.sol

```diff
@@ -260,7 +260,7 @@ contract Sales is DSMath { // Auctions
 		require(now > sales[sale].salex);
 		require(hasSecs(sale));
 		require(sha256(abi.encodePacked(sechs[sale].secD)) == sechs[sale].sechD);
-
+        sales[sale].taken = true;
```

Comment:
Move `sales[sale].taken = true;` above transfers to fix reentrancy attack vulnerability.